wgsl/index.bs

```diff
@@ -344,10 +339,64 @@ Examples of predeclared objects are:
 * built-in types.
 
 A declaration must not introduce a name when that identifier is
-already in scope at the start of the declaration.
-That is, shadow names are not allowed in [SHORTNAME].
+already in scope with the same end scope as another instance of that name.
+When an identifier is referenced and multiple declarations are in scope, the
+reference is resolved to the closest preceding in scope declaration.
```

Comment:
Earlier the spec says:

> The scope of a declaration is the set of program locations where a use of the declared identifier would denote its associated object.

That "denote" is no longer true.  The identifer denotes the object only if that's the resolved declaration. 

Recommend rewording that first one as:

> The scope of a declaration is the set of program locations where a use of the declared identifier potentially denotes its associated object.

Later:

> When an identifier is used in scope of one or more declarations for that name, the identifier will denote the object of the declaration appearing closest to that use. We say the identifier use resolves to that declaration.

(and make "resolves" a defined term.)

And add

> Note: A declaration always precedes its identifier's scope. Therefore the nearest in scope declaration of an identifier always precedes the use of the identifier.